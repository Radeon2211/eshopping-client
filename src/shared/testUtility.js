// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const historyPageNum = (pageNumber) => ({
  listen: jest.fn(),
  location: { search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push: jest.fn(),
});

export const propsPagination = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  maxQuantityPerPage: 2,
});
