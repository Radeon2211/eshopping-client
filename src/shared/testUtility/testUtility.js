// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';
import { v4 as uuidv4 } from 'uuid';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../../store/reducers/rootReducer';
import { productConditions } from '../constants';

export const defaultDeliveryAddress = {
  firstName: 'firstName',
  lastName: 'lastName',
  street: 'street',
  zipCode: 'zipCode',
  city: 'city',
  country: 'country',
  phone: 'phone',
};

export const defaultUserProfile = {
  ...defaultDeliveryAddress,
  _id: 'userId',
  username: 'username',
  email: 'email@domain.com',
  status: 'active',
  contacts: {
    email: true,
    phone: true,
  },
  createdAt: '2021-02-10T19:10:38.872Z',
};

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const createHistoryPageNumber = (pageNumber = 1, push = jest.fn()) => ({
  listen: jest.fn(),
  location: { pathname: '/products', search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push,
});

export const createPaginationProps = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  quantityPerPage: 2,
});

export const createProductItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
  quantitySold = 0,
  buyerQuantity = 0,
  description = '',
  condition = productConditions.NEW,
) => ({
  _id: productId || uuidv4(),
  name,
  price,
  quantity,
  quantitySold,
  buyerQuantity,
  description,
  condition,
  photo,
  seller: {
    username: sellerUsername,
  },
  __v: 0,
  createdAt: '2021-02-10T19:10:38.872Z',
  updatedAt: '2021-02-10T19:10:38.872Z',
});

export const createCartItem = (
  sellerUsername = 'username',
  quantity = 1,
  productId,
  price = 2,
  name = 'product name',
  productQuantity = 2,
  photo = false,
) => {
  const product = createProductItem(
    productId || uuidv4(),
    sellerUsername,
    productQuantity,
    price,
    name,
    photo,
  );

  return {
    _id: uuidv4(),
    quantity,
    product,
  };
};

export const createTransactionAndOrderProdItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
) => ({
  _id: productId || uuidv4(),
  name,
  photo,
  price,
  quantity,
  seller: {
    username: sellerUsername,
  },
});

export const createOrder = (
  products = [createTransactionAndOrderProdItem()],
  orderId = uuidv4(),
  sellerUsername,
  buyerUsername,
  overallPrice,
  createdAt,
  sellerEmail,
  sellerPhone,
) => {
  const seller = sellerUsername
    ? {
        username: sellerUsername,
        email: sellerEmail,
        phone: sellerPhone,
      }
    : null;

  const buyer = buyerUsername
    ? {
        username: buyerUsername,
      }
    : null;

  return {
    _id: orderId,
    seller,
    buyer,
    products,
    overallPrice,
    deliveryAddress: defaultDeliveryAddress,
    createdAt,
    updatedAt: createdAt,
    __v: 0,
  };
};

export const testStore = (initialAuth = {}, initialProduct = {}, initialUI = {}) => {
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
  const defaultState = createStoreWithMiddleware(rootReducer).getState();

  const finalState = {
    auth: {
      ...defaultState.auth,
      ...initialAuth,
    },
    product: {
      ...defaultState.product,
      ...initialProduct,
    },
    ui: {
      ...defaultState.ui,
      ...initialUI,
    },
  };
  const finalStore = createStoreWithMiddleware(rootReducer, finalState);

  return {
    store: finalStore,
    initialState: finalStore.getState(),
  };
};

export const checkReqMethodAndURL = (moxiosInstance, method, url, reqNumber = 0) => {
  if (reqNumber) {
    const requests = moxiosInstance.requests.__items;
    const prevRequest = requests[reqNumber - 1];
    return method === prevRequest.config.method && url === prevRequest.config.url;
  }
  const lastRequest = moxiosInstance.requests.mostRecent();
  return method === lastRequest.config.method && url === lastRequest.config.url;
};

export const createExpectedState = (
  initialState,
  expectedAuth = {},
  expectedProduct = {},
  expectedUI = {},
) => {
  const newState = {
    auth: {
      ...initialState.auth,
      ...expectedAuth,
    },
    product: {
      ...initialState.product,
      ...expectedProduct,
    },
    ui: {
      ...initialState.ui,
      ...expectedUI,
    },
  };

  return newState;
};

export const setUpStoreWithDefaultProfile = (auth = {}, product = {}, ui = {}) => {
  return testStore(
    {
      profile: defaultUserProfile,
      deliveryAddress: defaultDeliveryAddress,
      cart: [],
      ...auth,
    },
    product,
    ui,
  );
};
