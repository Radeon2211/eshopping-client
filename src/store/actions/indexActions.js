export {
  registerUser,
  loginUser,
  fetchProfile,
  logoutUser,
  fetchOtherUser,
  setOtherUser,
  updateUser,
  changeDeliveryAddress,
  deleteAccount,
  addAdmin,
  removeAdmin,
  sendAccountVerificationLink,
  resetPassword,
} from './authActions';

export {
  fetchCart,
  addCartItem,
  updateCartItem,
  clearCart,
  removeCartItem,
  setTransaction,
  goToTransaction,
  buyProducts,
} from './tradeActions';

export { setModal, deleteMessage, changeProductsPerPage } from './uiActions';

export {
  addProduct,
  editProduct,
  fetchProducts,
  fetchProductDetails,
  setProductDetails,
  deleteProduct,
} from './productActions';

export { fetchOrders, fetchOrderDetails, setOrderDetails } from './orderActions';
