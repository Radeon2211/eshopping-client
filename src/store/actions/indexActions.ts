export {
  registerUser,
  loginUser,
  fetchProfile,
  logoutUser,
  fetchOtherUser,
  setOtherUser,
  updateUser,
  changeEmail,
  changeDeliveryAddress,
  deleteAccount,
  addAdmin,
  removeAdmin,
  sendAccountVerificationLink,
  resetPassword,
} from './authActions/authActions';

export {
  fetchCart,
  addCartItem,
  updateCartItem,
  clearCart,
  removeCartItem,
  setTransaction,
  goToTransaction,
  buyProducts,
} from './tradeActions/tradeActions';

export { setMessage, setModal, changeProductsPerPage } from './uiActions/uiActions';

export {
  addProduct,
  editProduct,
  fetchProducts,
  fetchProductDetails,
  setProductDetails,
  deleteProduct,
} from './productActions/productActions';

export { fetchOrders, fetchOrderDetails, setOrderDetails } from './orderActions/orderActions';
