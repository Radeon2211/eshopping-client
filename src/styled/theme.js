const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#666',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#00A790',
    greenDark: '#009680',
    red: '#F22613',
  },
  shadows: {
    level1: '0 0 2px rgba(0, 0, 0, .1)',
    level2: '0 0 2px rgba(0, 0, 0, .3)',
    level3: '0 0 3px rgba(0, 0, 0, .3)',
    level4: '0 0 5px rgba(0, 0, 0, .3)',
    level5: '0 0 8px rgba(0, 0, 0, .3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Ubuntu, sans-serif',
    text: 'Lato, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
  },
};

export default theme;
