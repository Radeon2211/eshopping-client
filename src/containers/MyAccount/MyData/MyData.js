import React, { useCallback } from 'react';
import styled from 'styled-components';
import { useDispatch, useSelector } from 'react-redux';
import * as actions from '../../../store/actions/indexActions';
import * as modalTypes from '../../../store/actions/modalTypes';
import Panel from '../../../components/UI/Panel/Panel';
import Heading from '../../../components/UI/Heading/Heading';
import SingleInfo from './SingleInfo/SingleInfo';

const SC = {};
SC.Wrapper = styled.div`
  display: grid;
  grid-gap: ${({ theme }) => theme.spacings.level3};
  grid-template-columns: repeat(3, 1fr);

  @media only screen and (max-width: 37.5em) {
    grid-template-columns: repeat(2, 1fr);
  }
`;

const MyData = () => {
  const userProfile = useSelector((state) => state.auth.profile);

  const dispatch = useDispatch();
  const onSetModal = useCallback(
    (isModalOpen, modalContent) => dispatch(actions.setModal(isModalOpen, modalContent)),
    [dispatch],
  );

  let wrapperContent = null;
  if (userProfile) {
    const {
      firstName,
      lastName,
      username,
      email,
      street,
      zipCode,
      city,
      country,
      phone,
      contacts,
    } = userProfile;

    wrapperContent = (
      <>
        <SingleInfo
          name="First and last name"
          content={`${firstName} ${lastName}`}
          clickHandler={() => onSetModal(true, modalTypes.CHANGE_NAME)}
        />
        <SingleInfo
          name="Email"
          content={email}
          clickHandler={() => onSetModal(true, modalTypes.CHANGE_EMAIL)}
        />
        <SingleInfo name="Username" content={username} />
        <SingleInfo
          name="Address"
          content={{ street, zipCodeAndCity: `${zipCode} ${city}`, country }}
          clickHandler={() => onSetModal(true, modalTypes.CHANGE_ADDRESS)}
        />
        <SingleInfo
          name="Contacts"
          content={contacts}
          clickHandler={() => onSetModal(true, modalTypes.CHANGE_CONTACTS)}
        />
        <SingleInfo
          name="Phone number"
          content={phone}
          clickHandler={() => onSetModal(true, modalTypes.CHANGE_PHONE_NUMBER)}
        />
      </>
    );
  }

  return (
    <>
      <Heading variant="h3">My data</Heading>
      <Panel>
        <SC.Wrapper>{wrapperContent}</SC.Wrapper>
      </Panel>
    </>
  );
};

export default MyData;
