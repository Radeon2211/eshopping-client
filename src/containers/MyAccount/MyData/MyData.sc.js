import styled from 'styled-components';
import { Link } from 'react-router-dom';

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = styled.div`
  display: grid;
  grid-gap: ${({ theme }) => theme.spacings.level3};
  grid-template-columns: repeat(3, 1fr);

  & .change-passwd-btn {
    grid-area: 3 / 2 / 3 / 3;
  }

  & .actions {
    grid-column: 1 / -1;
  }

  @media only screen and (max-width: 37.5em) {
    grid-template-columns: repeat(2, 1fr);

    & .change-passwd-btn {
      grid-area: 4 / 1 / 4 / 3;
    }

    & .actions > * {
      flex: 1;
    }
  }
`;

export const PendingUserLogoutBtn = styled(Link)`
  @media only screen and (max-width: 37.5em) {
    & > * {
      width: 100%;
    }
  }
`;
