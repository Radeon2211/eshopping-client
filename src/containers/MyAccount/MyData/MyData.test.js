import React from 'react';
import { render, cleanup, screen, fireEvent, waitFor } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import MyData from './MyData';
import theme from '../../../styled/theme';
import { defaultUserProfile } from '../../../shared/testUtility/testUtility';
import * as actions from '../../../store/actions/indexActions';
import {
  defaultScrollToConfig,
  modalTypes,
  singleInfoNames,
  userStatuses,
} from '../../../shared/constants';

const mockStore = configureMockStore([thunk]);

const setUp = (profile, pushFn = jest.fn()) => {
  const store = mockStore({
    auth: { profile },
  });
  store.dispatch = jest.fn();

  const props = {
    history: {
      push: pushFn,
    },
  };

  return {
    ...render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <MyData {...props} />
        </ThemeProvider>
      </Provider>,
    ),
    store,
  };
};

afterEach(cleanup);

beforeAll(() => {
  window.scrollTo = jest.fn();
});

describe('<MyData />', () => {
  describe('check how renders', () => {
    it('should render everything correctly for non admin user with status active', async () => {
      const { asFragment } = setUp(defaultUserProfile);
      expect(asFragment()).toMatchSnapshot();
      await waitFor(() => {
        expect(document.title).toEqual('Your account data - E-Shopping');
      });
    });

    it('should render everything correctly for non admin user with status pending', () => {
      const { asFragment } = setUp({ ...defaultUserProfile, status: userStatuses.PENDING });
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render everything correctly for admin user', () => {
      const { asFragment } = setUp({ ...defaultUserProfile, isAdmin: true });
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('check behaviour after clicking', () => {
    it('should call correct functions after clicking in admin and active user content', () => {
      const { store } = setUp({ ...defaultUserProfile, isAdmin: true });

      fireEvent.click(screen.getByRole('button', { name: /add admin/i }));
      expect(store.dispatch).toHaveBeenNthCalledWith(1, actions.setModal(modalTypes.ADD_ADMIN));

      fireEvent.click(screen.getByRole('button', { name: /remove admin/i }));
      expect(store.dispatch).toHaveBeenNthCalledWith(2, actions.setModal(modalTypes.REMOVE_ADMIN));

      fireEvent.click(screen.getByTestId(`SingleInfo-${singleInfoNames.NAME}-btn`));
      expect(store.dispatch).toHaveBeenNthCalledWith(3, actions.setModal(modalTypes.CHANGE_NAME));

      fireEvent.click(screen.getByTestId(`SingleInfo-${singleInfoNames.EMAIL}-btn`));
      expect(store.dispatch).toHaveBeenNthCalledWith(4, actions.setModal(modalTypes.CHANGE_EMAIL));

      fireEvent.click(screen.getByTestId(`SingleInfo-${singleInfoNames.ADDRESS}-btn`));
      expect(store.dispatch).toHaveBeenNthCalledWith(
        5,
        actions.setModal(modalTypes.CHANGE_ADDRESS),
      );

      fireEvent.click(screen.getByTestId(`SingleInfo-${singleInfoNames.CONTACTS}-btn`));
      expect(store.dispatch).toHaveBeenNthCalledWith(
        6,
        actions.setModal(modalTypes.CHANGE_CONTACTS),
      );

      fireEvent.click(screen.getByTestId(`SingleInfo-${singleInfoNames.PHONE_NUMBER}-btn`));
      expect(store.dispatch).toHaveBeenNthCalledWith(
        7,
        actions.setModal(modalTypes.CHANGE_PHONE_NUMBER),
      );

      fireEvent.click(screen.getByRole('button', { name: /change password/i }));
      expect(store.dispatch).toHaveBeenNthCalledWith(
        8,
        actions.setModal(modalTypes.CHANGE_PASSWORD),
      );

      fireEvent.click(screen.getByRole('button', { name: /delete account/i }));
      expect(store.dispatch).toHaveBeenNthCalledWith(
        9,
        actions.setModal(modalTypes.DELETE_ACCOUNT),
      );
    });

    it('should call correct functions after clicking in pending user content', () => {
      const pushFn = jest.fn();
      const { store } = setUp({ ...defaultUserProfile, status: userStatuses.PENDING }, pushFn);

      fireEvent.click(screen.getByRole('button', { name: /send verification link/i }));
      expect(store.dispatch).toHaveBeenCalledWith(
        actions.setModal(modalTypes.SEND_VERIFICATION_LINK),
      );

      fireEvent.click(screen.getByRole('button', { name: /logout/i }));
      expect(pushFn).toHaveBeenCalledWith('/logout');
    });
  });

  describe('check useEffect()', () => {
    it('should call scrollTo() after mounting', () => {
      setUp();
      expect(window.scrollTo).toHaveBeenCalledWith(defaultScrollToConfig);
    });
  });
});
