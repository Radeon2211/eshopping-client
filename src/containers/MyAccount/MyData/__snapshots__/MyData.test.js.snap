// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyData /> check how renders should render everything correctly for admin user 1`] = `
<DocumentFragment>
  <h3
    class="sc-eCssSg eXQgPS"
  >
    My data
  </h3>
  <div
    class="sc-hKgILt eboeig"
    direction="column"
    spacing="3"
    width="100%"
    wrap="nowrap"
  >
    <section
      class="sc-dlfnbm dSatpG"
    >
      <div
        class="sc-bdfBwQ iLceJD"
      >
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Username"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Username
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            username
          </span>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Name"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Name
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            firstName lastName
          </span>
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            data-testid="SingleInfo-Name-btn"
            type="button"
          >
            Change
          </button>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Email"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Email
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            email@domain.com
          </span>
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            data-testid="SingleInfo-Email-btn"
            type="button"
          >
            Change
          </button>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Address"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Address
          </h4>
          <div
            class="sc-hKgILt cUIkdS"
            direction="column"
            spacing="1"
            width="100%"
            wrap="nowrap"
          >
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              street
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              zipCode city
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              country
            </span>
          </div>
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            data-testid="SingleInfo-Address-btn"
            type="button"
          >
            Change
          </button>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Contacts"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Contacts
          </h4>
          <div
            class="sc-hKgILt cUIkdS"
            direction="column"
            spacing="1"
            width="100%"
            wrap="nowrap"
          >
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              Email: visible
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              Phone number: visible
            </span>
          </div>
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            data-testid="SingleInfo-Contacts-btn"
            type="button"
          >
            Change
          </button>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Phone number"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Phone number
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            phone
          </span>
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            data-testid="SingleInfo-Phone number-btn"
            type="button"
          >
            Change
          </button>
        </div>
        <div
          class="sc-hKgILt kWUthc actions"
          direction="row"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            type="button"
          >
            Change password
          </button>
          <button
            class="sc-jSgupP hQkrdZ"
            color="red"
            type="button"
          >
            Delete account
          </button>
        </div>
        <div
          class="sc-hKgILt kWUthc actions"
          data-testid="MyData-admin-content"
          direction="row"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            type="button"
          >
            Add admin
          </button>
          <button
            class="sc-jSgupP hQkrdZ"
            color="red"
            type="button"
          >
            Remove admin
          </button>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`<MyData /> check how renders should render everything correctly for non admin user with status active 1`] = `
<DocumentFragment>
  <h3
    class="sc-eCssSg eXQgPS"
  >
    My data
  </h3>
  <div
    class="sc-hKgILt eboeig"
    direction="column"
    spacing="3"
    width="100%"
    wrap="nowrap"
  >
    <section
      class="sc-dlfnbm dSatpG"
    >
      <div
        class="sc-bdfBwQ iLceJD"
      >
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Username"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Username
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            username
          </span>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Name"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Name
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            firstName lastName
          </span>
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            data-testid="SingleInfo-Name-btn"
            type="button"
          >
            Change
          </button>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Email"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Email
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            email@domain.com
          </span>
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            data-testid="SingleInfo-Email-btn"
            type="button"
          >
            Change
          </button>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Address"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Address
          </h4>
          <div
            class="sc-hKgILt cUIkdS"
            direction="column"
            spacing="1"
            width="100%"
            wrap="nowrap"
          >
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              street
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              zipCode city
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              country
            </span>
          </div>
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            data-testid="SingleInfo-Address-btn"
            type="button"
          >
            Change
          </button>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Contacts"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Contacts
          </h4>
          <div
            class="sc-hKgILt cUIkdS"
            direction="column"
            spacing="1"
            width="100%"
            wrap="nowrap"
          >
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              Email: visible
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              Phone number: visible
            </span>
          </div>
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            data-testid="SingleInfo-Contacts-btn"
            type="button"
          >
            Change
          </button>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Phone number"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Phone number
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            phone
          </span>
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            data-testid="SingleInfo-Phone number-btn"
            type="button"
          >
            Change
          </button>
        </div>
        <div
          class="sc-hKgILt kWUthc actions"
          direction="row"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            type="button"
          >
            Change password
          </button>
          <button
            class="sc-jSgupP hQkrdZ"
            color="red"
            type="button"
          >
            Delete account
          </button>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`<MyData /> check how renders should render everything correctly for non admin user with status pending 1`] = `
<DocumentFragment>
  <h3
    class="sc-eCssSg eXQgPS"
  >
    My data
  </h3>
  <div
    class="sc-hKgILt eboeig"
    direction="column"
    spacing="3"
    width="100%"
    wrap="nowrap"
  >
    <section
      class="sc-dlfnbm dSatpG"
      data-testid="MyData-pending-user-content"
    >
      <span
        class="sc-fubCfw hUhCPF"
        color=""
        display="inline-block"
        overflow="visible"
        size="3"
        spacing="0px"
        transform=""
      >
        You need to activate your account to unlock all app functionalities. Verification link is active for 10 minutes. You can resend it below. If you do not activate your account within 1 hour, account will be deleted permanently.
      </span>
    </section>
    <section
      class="sc-dlfnbm dSatpG"
    >
      <div
        class="sc-bdfBwQ iLceJD"
      >
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Username"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Username
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            username
          </span>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Name"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Name
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            firstName lastName
          </span>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Email"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Email
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            email@domain.com
          </span>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Address"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Address
          </h4>
          <div
            class="sc-hKgILt cUIkdS"
            direction="column"
            spacing="1"
            width="100%"
            wrap="nowrap"
          >
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              street
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              zipCode city
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              country
            </span>
          </div>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Contacts"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Contacts
          </h4>
          <div
            class="sc-hKgILt cUIkdS"
            direction="column"
            spacing="1"
            width="100%"
            wrap="nowrap"
          >
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              Email: visible
            </span>
            <span
              class="sc-fubCfw osNBQ"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              Phone number: visible
            </span>
          </div>
        </div>
        <div
          class="sc-hKgILt gmWsbW"
          data-testid="SingleInfo-Phone number"
          direction="column"
          spacing="2"
          width="100%"
          wrap="nowrap"
        >
          <h4
            class="sc-eCssSg eLKVbs"
          >
            Phone number
          </h4>
          <span
            class="sc-fubCfw osNBQ"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            phone
          </span>
        </div>
        <div
          class="sc-hKgILt kWUthc actions"
          data-testid="MyData-pending-user-actions"
          direction="row"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            type="button"
          >
            Send verification link
          </button>
          <button
            class="sc-jSgupP biAJYh"
            color="blue"
            type="button"
          >
            Logout
          </button>
        </div>
        <div
          class="sc-hKgILt kWUthc actions"
          direction="row"
          spacing="3"
          width="100%"
          wrap="nowrap"
        >
          <button
            class="sc-jSgupP hQkrdZ"
            color="red"
            type="button"
          >
            Delete account
          </button>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
